include/cutlass/layout/permute.h

```diff
@@ -159,10 +156,10 @@ class Tensor4DPermute0213 {
     int i = row_init / D1;
 
     // After the Permute Op
-    col_permute_ = l + j * D3;
-    row_permute_ = k + i * D2;
+    Index col_permute = l + j * D3;
+    Index row_permute = k + i * D2;
 
-    return LongIndex(row_permute_) * LongIndex(stride_permute_) + LongIndex(col_permute_);
+    return LongIndex(row_permute) * LongIndex(stride_permute_) + LongIndex(col_permute);
```

Comment:
Question here: given that we know `[D0, D1, D2, D3]`, and the corresponding coordinate `[i, j, k, l]`, can we compute the final index using something like $l + D_3k + jD_2D_3j + iD_1D_2D_3$ ?